internal/cli/service.go

```diff
@@ -650,31 +646,92 @@ func (s Service) taskDefinitionsFromPaths(paths []string) ([]api.TaskDefinition,
 	return taskDefinitions, nil
 }
 
-// yamlFilePathsInDirectory returns any *.yml and *.yaml files in a given directory, ignoring any sub-directories.
-func (s Service) yamlFilePathsInDirectory(dir string) ([]string, error) {
-	paths := make([]string, 0)
+func (s Service) mintDirectoryEntries(dir string) ([]api.MintDirectoryEntry, error) {
+	mintDirectoryEntries := make([]api.MintDirectoryEntry, 0)
 
-	files, err := os.ReadDir(dir)
+	err := filepath.Walk(dir, func(pathInDir string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("error reading %q: %w", pathInDir, err)
+		}
+
+		mode := info.Mode()
+		permissions := mode.Perm()
+
+		var entryType string
+		switch mode.Type() {
+		case os.ModeDir:
+			entryType = "dir"
+		case os.ModeSymlink:
+			entryType = "symlink"
+		case os.ModeNamedPipe:
+			entryType = "named-pipe"
+		case os.ModeSocket:
+			entryType = "socket"
+		case os.ModeDevice:
+			entryType = "device"
+		case os.ModeCharDevice:
+			entryType = "char-device"
+		case os.ModeIrregular:
+			entryType = "irregular"
+		default:
+			if mode.IsRegular() {
+				entryType = "file"
+			} else {
+				entryType = "unknown"
+			}
+		}
```

Comment:
The only ones we really care about are dir and file - the rest are there just so I can provide a nicer error message about what we'll ultimately ignore